openlibrary/plugins/importapi/import_validator.py

```diff
@@ -1,36 +1,85 @@
-from typing import Annotated, Any, TypeVar
+from typing import Annotated, Any, Final, TypeVar
 
 from annotated_types import MinLen
-from pydantic import BaseModel, ValidationError
+from pydantic import BaseModel, ValidationError, model_validator
 
 T = TypeVar("T")
 
 NonEmptyList = Annotated[list[T], MinLen(1)]
 NonEmptyStr = Annotated[str, MinLen(1)]
 
+STRONG_IDENTIFIERS: Final = {"isbn_10", "isbn_13", "lccn"}
+
 
 class Author(BaseModel):
     name: NonEmptyStr
 
 
-class Book(BaseModel):
+class CompleteBookPlus(BaseModel):
+    """
+    The model for a complete book, plus source_records and publishers.
+
+    A complete book has title, authors, and publish_date. See #9440.
+    """
+
     title: NonEmptyStr
     source_records: NonEmptyList[NonEmptyStr]
     authors: NonEmptyList[Author]
     publishers: NonEmptyList[NonEmptyStr]
     publish_date: NonEmptyStr
 
 
+class StrongIdentifierBookPlus(BaseModel):
```

Comment:
If wanted more descriptive terminology to reflect the intent of this check, I was using 'differentiatable' to mean that we can determine what specific _edition_ the import data relates to, and that captures the complete OR has a at least one hard  identifier.